src/api.rs

```diff
@@ -489,35 +491,42 @@ impl Config {
       first_pass_data: FirstPassData {
         frames: Vec::new(),
       },
+      pool,
+      },
+      config: self.enc.clone(),
     }
   }
 }
 
-pub struct Context<T: Pixel> {
-  //    timebase: Rational,
+pub struct ContextInner<T: Pixel> {
   frame_count: u64,
   limit: u64,
   pub(crate) idx: u64,
   frames_processed: u64,
   /// Maps frame *number* to frames
   frame_q: BTreeMap<u64, Option<Arc<Frame<T>>>>, //    packet_q: VecDeque<Packet>
   /// Maps frame *idx* to frame data
-  frame_data: BTreeMap<u64, FrameInvariants<T>>,
+  frame_invariants: BTreeMap<u64, FrameInvariants<T>>,
   /// A list of keyframe *numbers* in this encode. Needed so that we don't
-  /// need to keep all of the frame_data in memory for the whole life of the encode.
+  /// need to keep all of the frame_invariants in memory for the whole life of the encode.
   keyframes: BTreeSet<u64>,
   /// A storage space for reordered frames.
   packet_data: Vec<u8>,
   segment_start_idx: u64,
   segment_start_frame: u64,
   keyframe_detector: SceneChangeDetector<T>,
-  pub config: Config,
+  pub(crate) config: EncoderConfig,
   rc_state: RCState,
   maybe_prev_log_base_q: Option<i64>,
   pub first_pass_data: FirstPassData,
   pool: rayon::ThreadPool,
 }
 
+pub struct Context<T: Pixel> {
+  inner: ContextInner<T>,
+  config: EncoderConfig,
```

Comment:
What is the use case for having EncoderConfig in both Context and ContextInner? Is it avoidable somehow?

Reply:
It is avoidable, but in the next set. Keeping it around reduces the amount of code changes.